src/java/boa/BoaTup.java

```diff
@@ -0,0 +1,11 @@
+package boa;
```

Comment:
What is the relation between BoaTup and BoaTuple, a class that already exists?

Reply:
BoaTup is used to read the training data.

Reply:
Please see src/java/boa/types/BoaTuple.java. Can that existing type serve this purpose?

Reply:
I'm not even sure this is being used.

There are no classes inheriting from this interface.  There are no instances of this type being created anywhere.

I think this file, and other code referencing it (like the functions that accept a BoaTup as argument), can probably be removed?

Reply:
Currently, I am still using BoaTup for the training data. I have added the function which uses BoaTup. This function will be used to classify the data from trained model.